discussion/serializers.py

```diff
@@ -22,7 +22,9 @@ def get_user(self, obj):
         return serializer.data
 
     def get_tags(self, obj):
-        return ['Python', 'Django', 'Postman', 'API']
+        tags = obj.tags.all()
+        serializer = SkillTagSerializer(tags, many=True)
+        return serializer.data
```

Comment:
If you are going to return the serialized data you can just delete this method and use the serializer directly:
```
class DiscussionSerializer(serializers.ModelSerializer):
    user = serializers.SerializerMethodField(read_only=True)
    tags = SkillTagSerializer(many=True, read_only=True)
```
But make sure that this doesn't  break the frontend, because this will return a list of objects instead of a simple list as before. Maybe the method is needed but to return a list of values and not a serialized queryset.

Reply:
Yeah, I see thanks for the feedback. About the data returned, any idea on how to return the data in this format [ tag1, tag2 ]
```json
# data presently returned
{
"name": "tag1"
},
{
"name": "tag2"
}
```

Reply:
Maybe the frontend could be updated to use the new structure, or you could add the method again and do something like:
```
def get_tags(self, obj):
    tags = obj.tags.values_list('name', flat=True)
    return tags
```
I think that should return the simple list.

Reply:
Thanks

Reply:
You can just simply do this 

```
tags = obj.tags.all()
return tags
```

Reply:
I'm using value list to return the tags in the same format as it was i.e [ 'python', 'postman' ]

Or should I change it as you mentioned?
That would return
```json

```

Reply:
`.all()` will return a list so in serializer it will become an array !

Reply:
Alright 👍